src/sb_gazebo/robots/rover_2018.xacro

```diff
@@ -0,0 +1,287 @@
+        />
+    <xacro:property name="primary_rocker_arm_y_offset" value="0"/>
+    <xacro:property name="primary_rocker_arm_z_offset" 
+        value="${- sin(primary_rocker_arm_pitch)*primary_rocker_arm_length/2}"/>
+
+    <xacro:property name="primary_wheel_axle_length" value="0.1"/>
+    <xacro:property name="primary_wheel_axle_radius" value="0.015"/>
+    <xacro:property name="primary_wheel_axle_x_offset" 
+        value="${sin(primary_rocker_arm_pitch)*primary_rocker_arm_length/2}"/>
+    <xacro:property name="primary_wheel_axle_y_offset" value="${primary_wheel_axle_length/2}"/>
+    <xacro:property name="primary_wheel_axle_z_offset" 
+        value="${cos(primary_rocker_arm_pitch)*primary_rocker_arm_length/2}"/>
+
+    <xacro:property name="primary_wheel_length" value="0.15"/>
+    <xacro:property name="primary_wheel_radius" value="0.17"/>
+    <xacro:property name="primary_wheel_x_offset" value="0"/>
+    <xacro:property name="primary_wheel_y_offset" value="${primary_wheel_length/2}"/>
+    <xacro:property name="primary_wheel_z_offset" value="0"/>
+
+    <xacro:property name="tower_upright_x_offset" value="${chassis_y_size/2}"/>
+    <xacro:property name="tower_upright_height" value="0.5"/>
+    <xacro:property name="tower_upright_x_size" value="0.02"/>
+    <xacro:property name="tower_upright_y_size" value="0.02"/>
+    <xacro:property name="tower_upright_z_offset" value="${tower_upright_height/2 + chassis_z_size/2}"/>
+
+    <xacro:property name="camera_size" value="0.05"/>
+    <xacro:property name="camera_pitch" value="${pi/4}"/>
+    <xacro:property name="camera_x_offset" value="0.03"/>
+    <xacro:property name="camera_z_offset" value="${tower_upright_height/2}"/>
+
+    <!-- JOINTS AND LINKS
+         This section uses all the values calculated above to actually create
+         all the links and joints of the robot
+    -->
+
+    <!-- Footprint of the robot (because gazebo's base_link can't have interia) -->
+    <link name='base_link'/>
+    <joint name="base_joint" type="fixed">
+        <parent link="base_link"/>
+        <child link="chassis"/>
+        <origin xyz="
+            ${chassis_x_pos}
+            ${chassis_y_pos}
+            ${chassis_z_pos}
+            " 
+            rpy="0 0 0"
+            />
+    </joint>
+    <box_link 
+        name="chassis"
+        x_pos="0"
+        y_pos="0"
+        z_pos="0"
+        x_size="${chassis_x_size}"
+        y_size="${chassis_y_size}"
+        z_size="${chassis_z_size}"
+        roll="0" 
+        pitch="0" 
+        yaw="0"
+        mass="10"
+        material_name="grey"
+        />
+
+    <xacro:macro name="primary_rocker" params="lr_prefix y_reflect">
+        <!-- TODO: Make this revolute model with a spring to resist rotation (but allow it) instead of just being fixed like it currently is -->
+        <joint name="${lr_prefix}_primary_rocker_to_chassis" type="revolute">
```

Comment:
A comment explaining what lr_prefix means would be good.

Reply:
Expanded all to `left_right_prefix` and added javadoc style comment to each macro.